frontend/src/components/NewSourceForm.js

```diff
@@ -22,13 +22,29 @@ class NewSourceForm extends React.Component {
 
     addNewSource = () => {
         this.setLoading(true);
-        const url = document.getElementById(this.state.newSourceUrlId).value;
+        let url;
+        if (!this.props.source) {
+            url = document.getElementById(this.state.newSourceUrlId).value;
```

Comment:
General comment: make sure to use the word "item" instead of "source" now (unless we are actually dealing with a source, which is unlikely in most cases).

Reply:
Fixed in bca252eee660983e4c1d65e493bf25542b5c7fcf